packages/apps/docs/src/components/Markdown/Heading/Heading.tsx

```diff
@@ -93,12 +48,101 @@ export const TaggedHeading: FC<IProp> = ({
 
   const slug = createSlug(slugInputStr, index, parentTitle);
 
-  return (
-    <StyledHeader as={as} variant={variant ?? as}>
-      {children}
-      <StyledLinkIcon id={slug} href={`#${slug}`}>
-        <SystemIcons.Link />
-      </StyledLinkIcon>
-    </StyledHeader>
+  // vanilla-extract doesn't support styling subcomponents
+  // globalStyle doesn't support simple pseudo selectors
+  // only way around the hover behavior is via state + variants
+  const [linkHoverVariant, setLinkHoverVariant] = useState<boolean>(false);
+
+  const onMouseOver = (): void => {
+    setLinkHoverVariant(true);
+  };
+
+  const onMouseLeave = (): void => {
+    setLinkHoverVariant(false);
+  };
+
+  const ArticleLink = (
+    <a
+      className={linkHoverVariant ? headerIconLinkHovered : headerIconLink}
+      id={slug}
+      href={`#${slug}`}
+    >
+      <SystemIcon.Link />
+    </a>
   );
+  switch (as) {
+    case 'h2':
+      return (
+        <h2
+          onMouseLeave={onMouseLeave}
+          onMouseOver={onMouseOver}
+          onFocus={() => {}}
+          className={headerVariants[variant ?? as]}
+        >
+          {children}
+          {ArticleLink}
+        </h2>
+      );
+    case 'h3':
+      return (
```

Comment:
why are you changing into switch case here? can't we re-use Heading component and achieve the same?

Reply:
We need to apply the `headerVariants` class to them, which is not an accepted parameter by `Heading`. Should we change the component instead?

Reply:
The articles use <h1> and <h2> quite often. Without the resize (which makes them smaller than the default) the titles are less aggressive. That said, the resize is here because it was part of the stitches implementation, not sure what motivated it, originally

Reply:
by resize do you mean the reponsive fontSize changes on the Heading component styles?

Reply:
So I think it's fine to have a separate heading component for this if the styles really are meant to differ greatly, but I think this implementation can be improved. For example, the hover states and focus states can be applied via css rather than passing in props

Reply:
The hover/focus states need to alter the style of`ArticleLink` but are bound to the `Heading` atop of them. Do we have a way to do that in vanilla-extract? Even with `globalStyle` we cannot use pseudo-selectors like `:hover`:
https://vanilla-extract.style/documentation/global-api/global-style/

Reply:
looking at your code, it looks like the only style difference on hover is the opacity. You believe you can achieve the same effect by using variables. You can define an `articleLinkOpacity` var and set it to it's default opacity in the normal heading styles, and to the hover opacity through the hover selector. Then you can use this variable in the css definition for the articleLink. Hope this helps and let me know if you have any questions

Reply:
You're absolutely right, didn't even think of using variables for this one!

Just pushed some changes using `useVariable`, works like a charm :smiley: